src/PrestaShopBundle/Entity/Repository/NormalizeFieldTrait.php

```diff
@@ -48,6 +48,25 @@ protected function castNumericToInt($rows)
     }
 
     /**
+     * @param $rows
+     * @return mixed
+     */
+    protected function castIdsToArray($rows)
+    {
+        $castIdentifiersToArray = function (&$columnValue, $columnName) {
+            if ($this->shouldCastToInt($columnName)) {
+                $columnValue = explode(',', $columnValue);
+            }
+        };
+
+        array_walk($rows, function (&$rowColumns) use ($castIdentifiersToArray) {
```

Comment:
Have you checked [array-walk-recursive()](http://php.net/manual/en/function.array-walk-recursive.php)?

Or maybe a `foreach()` would be enough?